### PR TITLE
Skip macOS other-app data bundles (.photoslibrary) when scanning photo folders

### PR DESCRIPTION
**Pull request by jss367**
## Problem

Vireo repeatedly popped the macOS dialog **\"Vireo.app would like to access data from other apps\"** — clicking **Allow** had no lasting effect; it just came back.

### Root cause

The default scan root `~/Pictures` contains `Photos Library.photoslibrary` — Apple Photos' private data bundle. The scanner's `os.walk` (`vireo/scanner.py`) never pruned directories, so it descended straight into that bundle. On macOS Sequoia, reading inside **another app's data area** triggers the `kTCCServiceSystemPolicyAppData` consent prompt. Because the library holds thousands of files, the walk re-trips the prompt on every file it touches — so Allow never makes it stop. The contents are app-managed derivatives Vireo would never ingest anyway.

The installed app is correctly Developer ID–signed and notarized; this was never a signing problem.

## Fix

Add `prune_scan_dirs()` / `is_excluded_scan_dir()` to `image_loader.py` and apply them at every walker that traverses user-chosen roots:

- `scanner.scan` (the reported bug) — prunes `dirnames` in place, logs what it skipped
- `new_images` — same, so the "new images" banner isn't inflated by library contents
- `audit` — untracked-files check (converted `rglob`→`os.walk` to allow pruning) and stray-sidecar check
- `ingest` — recursive source discovery (converted `rglob`→`os.walk`), so picking `~/Pictures` as an import source doesn't walk the whole library

Excluded (case-insensitive): `*.photoslibrary`, `*.photolibrary`, `*.migratedphotolibrary`, `*.aplibrary`, `*.migratedaplibrary`, and `Photo Booth Library`.

## Tests

- Helper unit tests in `test_image_loader.py` (suffix/exact/case matching, in-place prune, no-op)
- Integration test in `test_scanner.py`: images inside a `.photoslibrary` / `Photo Booth Library` are ignored while real sibling photos are still discovered

Full project suite: **1170 passed, 1 skipped**. Affected-module suite: **229 passed**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented macOS app-managed photo library bundles (e.g., `*.photoslibrary` / `*.aplibrary`) from being traversed during browsing, scanning, ingesting, importing, rescanning, and photo counting, with early validation and clear `400` errors across relevant endpoints (including pipeline `source`/`sources`).
  * Improved traversal safety by skipping excluded bundles (including via symlinks) and ignoring broken symlink entries to avoid incorrect audit results.

* **Tests**
  * Added regression coverage for bundle exclusion, early-validation ordering, audit untracked/sidecar behavior, symlink edge cases, and job/rescan/browse API responses.

### CODE REVIEW

**Comment by jss367** [comment]
@codex review

File: vireo/audit.py [hunk 1]
```diff
@@ -146,22 +145,20 @@ def check_untracked(db, root_paths):
...
+        # to stop descending and would walk the whole Photos library.
+        for dirpath, dirnames, filenames in os.walk(root):
+            prune_scan_dirs(dirnames)
+            for name in filenames:
+                if name.startswith(".") or os.path.splitext(name)[1].lower() not in SUPPORTED_EXTENSIONS:
+                    continue
+                full = os.path.join(dirpath, name)
+                if full in known_paths:
+                    continue
+                untracked.append({"path": full, "folder": dirpath})
```
[the remark this answers is not in the document]
Comment (jss367) [reply vireo/audit.py:185]: Good catch — fixed in a369c5d. Restored the `os.path.isfile(full)` guard the `Path.rglob`→`os.walk` conversion dropped, so dangling symlinks with image extensions are no longer reported untracked (matching `scanner.scan`'s behavior). Added a regression test. 👍

**Comment by jss367** [comment]
@codex review

**Comment by jss367** [comment]
@codex
